tests/impl.cpp

```diff
@@ -324,6 +324,16 @@ float compord(float a, float b)
     return ret;
 }
 
+double compord(double a, double b)
+{
+    double ret;
+
+    bool isNANA = !(a == a);
+    bool isNANB = !(b == b);
```

Comment:
Use `isNAN()` like `float compord(float, float)` does.
Or maybe we can just simplify it as
```c++
ret = (!isNAN(a) && !isNAN(b)) ? getNAN() : 0.0f;
```